### PR TITLE
add app platform collector + deploy button

### PR DESCRIPTION
**Pull request by circa10a**
@metalmatze  Addresses #15 

- This allows the monitoring of applications deployed on the digitalocean app platform as well as their deployment statuses.
  - Updated dependencies as a result(needed new godo client)
- I've also added a "deploy to digitalocean" button that allows folks to easily deploy the exporter on the app platform itself(tested using my own fork).
- 
Screen Shot 2021-01-04 at 1 29 28 PM

### CODE REVIEW

File: .do/deploy.template.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      github:
+        repo: metalmatze/digitalocean_exporter
+        branch: master
+      http_port: 9212
+      health_check:
+        http_path: /
+      envs:
+        - key: DIGITALOCEAN_TOKEN
+          value: CHANGE_ME
+          type: SECRET
```
Comment (metalmatze) [inline .do/deploy.template.yaml:15]: I'm not really sure I actually want every cloud provider to start adding these files to each repository...
Is it possible to make it work without the need for this extra file?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
-# DigitalOcean Exporter [![Build Status](https://cloud.drone.io/api/badges/metalmatze/digitalocean_exporter/status.svg)](https://cloud.drone.io/metalmatze/digitalocean_exporter)
+# DigitalOcean Exporter [![Deploy to DO](https://mp-assets1.sfo2.digitaloceanspaces.com/deploy-to-do/do-btn-blue.svg)](https://cloud.digitalocean.com/apps/new?repo=https://github.com/metalmatze/digitalocean_exporter/tree/master)
```
Comment (metalmatze) [inline README.md:1]: This button certainly is too big to go to the very top of the README.
I'd prefer this becoming another section under "Installation" in the README.

**Review by metalmatze — CHANGES_REQUESTED** [review]

File: .do/deploy.template.yaml (hunk 1 above)
Comment (metalmatze) [reply .do/deploy.template.yaml:15]: I'd make an exception here as this exporter is directly related to DigitalOcean though. Just funny.

Comment (circa10a) [reply .do/deploy.template.yaml:15]: Unfortunately, this seems to be the only supported method for deploying on the app platform

File: README.md (hunk 2 above)
Comment (circa10a) [reply README.md]: Absolutely, fixed.

Comment (circa10a) [reply README.md]: Screen Shot 2021-01-04 at 1 28 34 PM

Comment (circa10a) [reply README.md]: Screen Shot 2021-01-04 at 1 29 28 PM

File: .do/deploy.template.yaml (hunk 1 above)
Comment (circa10a) [reply .do/deploy.template.yaml:15]: [DO doc for reference](https://www.digitalocean.com/docs/app-platform/how-to/add-deploy-do-button/)

**Comment by circa10a** [comment]
@metalmatze I was hoping to have this addition submitted for the digitalocean hackathon which ends on Jan 10. Do you think you would be able to review/test before then?

**Comment by circa10a** [comment]
@metalmatze any updates?

**Comment by circa10a** [comment]
@metalmatze anymore thoughts on this change?

**Comment by circa10a** [comment]
Hey @metalmatze any updates on this?

**Comment by metalmatze** [comment]
No update really, sorry. 
I guess we can move forward with this.

Thinking about this once more, does it actually make sense to deploy this as DigitalOcean App? Did you try scraping it with a Prometheus?

**Comment by circa10a** [comment]
I think it does in the sense that it gives users a chance to try it out. Maybe the auth integration would be a nice enhancement if that gets supported natively on the app platform. If you prefer, I could remove the deploy bits and have this PR just scoped towards the collector.

I did deploy and use this functionality with prometheus. Here's a write up I did https://dev.to/circa10a/monitoring-the-digitalocean-app-platform-with-the-digitalocean-app-platform-4971

**Comment by metalmatze** [comment]
Alright. Cool!
Well, I hope users know what they are doing --- famous last words.

**Review by metalmatze — APPROVED** [review]

**Pull request merged**